src/ssh_client.py

```diff
@@ -0,0 +1,77 @@
+import paramiko
+import logging
+
+from src.base_client import BaseShellClient
+
+
+class SSHClient(BaseShellClient):
+    def __enter__(self):
+        try:
+            self.client = paramiko.SSHClient()
+            self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            self.client.connect(self.host, port=self.port, username=self.username, password=self.password)
+            logging.info(f"Connected to {self.host} successfully.")
+        except Exception as e:
+            logging.error(f"Failed to connect to {self.host}: {e}")
```

Comment:
Raise exception if connection was not established. And futher you don't  need to check if self.client is not None